app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt

```diff
@@ -0,0 +1,59 @@
+package com.github.se.cyrcle.model.user
+
+/**
+ * A type alias for the amount of coins a user has. Allow more flexibility in the future if we want
+ * to change the type of the coins.
+ */
+typealias Coin = Int
+
+/**
+ * A class that represent a [Coin] container (hence the name), which can be used to store and
+ * manipulate the amount of coins a user has.
+ */
+class Wallet(private var coins: Coin = 0) {
+
+  /**
+   * Add amount of coins to the wallet.
+   *
+   * @param amount the amount of coins to add. Must be positive.
+   */
+  fun creditCoins(amount: Coin) {
+    if (amount < 0) throw IllegalArgumentException("Amount must be positive")
+    coins += amount
+  }
+
+  /**
+   * Remove amount of coins to the wallet. This function does not check if the wallet is solvable.
+   *
+   * @param amount the amount of coins to remove. Must be positive.
+   */
+  fun debitCoins(amount: Coin) {
+    if (amount < 0) throw IllegalArgumentException("Amount must be positive")
+    coins -= amount
+  }
+
+  /**
+   * Return whether the user is solvable. A user is solvable if, after debiting the amount, the
+   * amount of coins in the wallet is still superior to the credit threshold. This function does not
+   * modify the value of the coins in the wallet, it's a simple projection.
+   */
+  fun isSolvable(amount: Coin, creditThreshold: Coin): Boolean {
+    if (amount < 0) throw IllegalArgumentException("Amount must be positive")
+    return (coins - amount) > creditThreshold
+  }
+
+  /**
+   * Return the amount of coins currently in the mallet. WARNING: This method does not modify the
+   * value of the coins in the wallet.
+   */
+  fun getCoins(): Coin {
+    return coins
+  }
+
+  /** Creates an empty [Wallet]. It contains no coins. */
+  companion object {
```

Comment:
I'm not sure if the `companion object` is realy necesary, we already have a default argument for that?

Reply:
It is maybe a bit overkill, but as the feature isn't well define in the project, I preferred doing a bit too much than too little (image if, by default, users have a welcome gift, then `Wallet(coin = 0)` make less sens.

Reply:
Right... that makes sense!